src/Codegen/Generator.php

```diff
@@ -16,11 +16,14 @@ class Generator
 {
     protected EndpointConfig $endpointConfig;
 
+    protected string $configFile;
+
     protected string $endpointName;
 
-    public function __construct(EndpointConfig $endpointConfig, string $endpointName)
+    public function __construct(EndpointConfig $endpointConfig, string $configFile, string $endpointName)
```

Comment:
Since config file and endpoint name frequently travel together, it might be easier to define a compound data structure:

```php
class EndpointReference {
    public string $configFile;
    public string $endpointName;
}
```